src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -367,6 +371,15 @@ public void updateUser(UserImpl userObj, DataObject user)
                     jda, responseNumber,
                     userObj, oldAvatar));
         }
+        
+        if (oldFlags != newFlags)
+        {
+            userObj.setFlags(newFlags);
+            jda.handleEvent(
+                    new UserUpdateFlagsEvent(
+                        jda, responseNumber,
+                        userObj, oldFlags));
```

Comment:
Doesn't this accept an `EnumSet` now?

Reply:
Should I change it to `User.UserFlag.getFlags(oldFlags)` or should I use a different approach?